pyccel/ast/variable.py

```diff
@@ -168,6 +172,10 @@ def __init__(
             raise TypeError('is_optional must be a boolean.')
         self._is_optional = is_optional
 
+        if not isinstance(is_private, bool):
+            raise TypeError('is_optional must be a boolean.')
```

Comment:
fix the raised error to `'is_private must be a boolean.'`

Reply:
Fixed 5041be14